src/Sylius/Component/Promotion/Repository/PromotionCouponRepositoryInterface.php

```diff
@@ -32,4 +33,12 @@ public function createQueryBuilderByPromotionId($promotionId);
      * @return int
      */
     public function countByCodeLength($codeLength);
+
+    /**
+     * @param string $promotionCode
+     * @param string $code
+     *
+     * @return PromotionInterface
+     */
+    public function findOneByPromotionCodeAndCode($promotionCode, $code);
```

Comment:
While I agree with @Zales0123's argument regarding the order of params in the routing, I think we should remember that code and routing are separate things (unless you are Sylius YAML Developer (tm)). So let's make the method name more clear and go with ``findOneByCodeAndPromotionCode``. 👍